src/uniforms/bind.rs

```diff
@@ -50,8 +50,6 @@ impl<U> UniformsExt for U where U: Uniforms {
             if visiting_result.is_err() { return; }
 
             if let Some(uniform) = program.get_uniform(name) {
-                assert!(uniform.size.is_none(), "Uniform arrays not supported yet");
```

Comment:
Please add `debug_assert!(uniform.size.is_none())`, since `size` should always be `None`. Also add a `TODO: remove the size member` either here or in the struct declaration.